pkgs/sdk/server-ai/src/LdAiClient.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.Json;
+using LaunchDarkly.Sdk.Server.Ai.Adapters;
+using LaunchDarkly.Sdk.Server.Ai.Config;
+using LaunchDarkly.Sdk.Server.Ai.DataModel;
+using LaunchDarkly.Sdk.Server.Ai.Interfaces;
+using LaunchDarkly.Sdk.Server.Interfaces;
+using Mustache;
+
+namespace LaunchDarkly.Sdk.Server.Ai;
+
+/// <summary>
+/// The LaunchDarkly AI client. The client is capable of retrieving AI configurations from LaunchDarkly,
+/// and generating events specific to usage of the AI configuration when interacting with model providers.
+/// </summary>
+public sealed class LdAiClient
```

Comment:
Typically we should have an interface on this so the customer can mock it. (And all the compatibility inconveniences with that.)

Reply:
Added an `ILdAiClient` and also changed the ModelConfig return to an `ILdAiConfigTracker`.